File: TopCoder/src/main/java/com/util/FizzBuzz.java
package com.util;

import java.util.LinkedList;
import java.util.List;

//https://leetcode.com/problems/fizz-buzz/
public class FizzBuzz {
    public List<String> fizzBuzz(int n) {
        List<String> retVal = new LinkedList<>();
        for (int i = 1 ; i <= n ; i++) {
           if ( i % 3 == 0 && i % 5 == 0) {
               retVal.add("FizzBuzz");
           } else if (i % 3 == 0) {
               retVal.add("Fizz");
           } else if (i % 5 == 0) {
               retVal.add("Buzz");
           } else {
               retVal.add(i+"");
           }
        }
        return retVal;
    }

    public static void main(String[] args) {
        int n = 15;
        List<String> res = new FizzBuzz().fizzBuzz(n);
    }
}

File: TopCoder/python/misc/FizzBuzz.py
class Solution:
    def fizzBuzz(self, n):
        """
        :type n: int
        :rtype: List[str]
        """
        i = 1
        retval = []
        while i <= n :
            if i % 3 == 0 and i % 5 == 0:
                retval.append("FizzBuzz")
            elif i % 3 == 0:
                retval.append("Fizz")
            elif i % 5 == 0:
                retval.append("Buzz")
            else:
                retval.append(str(i))
            i = i + 1
        return retval

sol = Solution()
ret = sol.fizzBuzz(1)
print(ret)